[ZEPPELIN-2396] eliminate the 'ctrl/command+L' keyboard shortcut

soralee opened the pull request:
### What is this PR for?
This PR is for that eliminate the `Ctrl/Command+L` shortcut keyboard. This function is what the ace editor defined.

### What type of PR is it?
[Improvement | Document]

### What is the Jira issue?
* [ZEPPELIN-2396](https://issues.apache.org/jira/browse/ZEPPELIN-2396)

### How should this be tested?
* **Improvement** - Press `Ctrl + L` key and check if showing the dialog or not
* **Document** - Run document development mode and check [this document](http://localhost:4000/install/upgrade.html#upgrading-from-zeppelin-07-to-08)

### Screenshots (if appropriate)
[Before]
![image](https://cloud.githubusercontent.com/assets/8110458/24990819/8529312a-2051-11e7-8849-e00803310752.png)

[After]
![z_zeppelin-2396](https://cloud.githubusercontent.com/assets/8110458/24990966/9764e108-2052-11e7-9387-560f9d587782.gif)



[Document]
![image](https://cloud.githubusercontent.com/assets/8110458/25040847/e772e024-2146-11e7-9ded-322c589b424b.png)



### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes, I did. Please review [this document](http://localhost:4000/install/upgrade.html#upgrading-from-zeppelin-07-to-08) if you run document development mode (localhost:4000)

1ambda commented:
Awesome! It was really annoying. Let me test and comment soon.

1ambda commented:
@soralee I think we need to add this change into https://github.com/apache/zeppelin/blob/master/docs/install/upgrade.md

1ambda commented:
@soralee Doesn't work for OSX users since the binding is `CMD + L` not `CTRL + L` in OSX.

soralee commented:
@1ambda Thanks for review! 
Yeah, it's make sense. Let me update `command+L` and the document

soralee commented:
I added `command+L` and changed `Alt` and `Option` feature to flexibly remove some binding shortcut key.
And, I updated the document as your comment.
Could you review again @1ambda ?

1ambda commented on docs/install/upgrade.md:
```diff
@@ -60,4 +60,4 @@ So, copying `notebook` and `conf` directory should be enough.
  - From 0.8, we recommend to use `PYSPARK_PYTHON` and `PYSPARK_DRIVER_PYTHON` instead of `zeppelin.pyspark.python` as `zeppelin.pyspark.python` only effects driver. You can use `PYSPARK_PYTHON` and `PYSPARK_DRIVER_PYTHON` as using them in spark.
- 
\ No newline at end of file
+ - From 0.8, depending on your device, the keyboard shortcut `Ctrl-L` or `Command-L` is not supported. 
```
It would be nice if we add what `CTRL+L` was.

soralee replied:
Thanks for suggestions. I just agonized the showing form about what kind of keyboard shortcut is more clear for us, because in my case, `command-l` is good as well. :smile: 
Anyway, I'll rewrite this!

1ambda commented on zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js:
```diff
@@ -709,19 +709,27 @@ function ParagraphCtrl($scope, $rootScope, $route, $window, $routeParams, $locat
-      $scope.editor.commands.bindKey('ctrl-alt-t', null);
+      var isOption = $rootScope.isMac? 'option' : 'alt';
+
+      $scope.editor.commands.bindKey('ctrl-'+isOption+'-n.', null);
```
MINOR: Can we add space between `+` for readability?

soralee replied:
Sure, let me rewrite this. Thanks!

1ambda commented:
Works as described. LGTM except 2 comments I left

soralee commented:
@1ambda I just updated it!

1ambda commented:
LGTM

Leemoonsoo commented:
LGTM and merge to master if no further discussions
